service/controller/prometheus.go

```diff
@@ -39,12 +34,40 @@ func NewPrometheus(config PrometheusConfig) (*Prometheus, error) {
 	if config.K8sClient == nil {
 		return nil, microerror.Maskf(invalidConfigError, "%T.K8sClient must not be empty", config)
 	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	if config.ConfigMapKey == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.ConfigMapKey must not be empty", config)
+	}
+	if config.ConfigMapName == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.ConfigMapName must not be empty", config)
+	}
+	if config.ConfigMapNamespace == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.ConfigMapNamespace must not be empty", config)
+	}
+	if config.CertComponentName == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.CertComponentName must not be empty", config)
+	}
+	if config.CertDirectory == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.CertDirectory must not be empty", config)
+	}
+	if config.CertNamespace == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.CertNamespace must not be empty", config)
+	}
+	if config.CertPermission == 0 {
+		return nil, microerror.Maskf(invalidConfigError, "%T.CertPermission must not be empty", config)
+	}
+	if config.PrometheusAddress == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.PrometheusAddress must not be empty", config)
+	}
```

Comment:
Some standard checks.

Reply:
These shouldn't be in the controller IMO.